lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -756,6 +761,77 @@ void TypeLoweringVisitor::visitStmt(ConnectOp op) {
   opsToRemove.push_back(op);
 }
 
+void TypeLoweringVisitor::recursivePartialConnect(Value a, FIRRTLType aType,
+                                                  Value b, FIRRTLType bType,
+                                                  Twine suffix, bool aFlip) {
+  TypeSwitch<FIRRTLType>(aType)
+      .Case<BundleType>([&](auto aType) {
+        auto bBundle = bType.dyn_cast_or_null<BundleType>();
+        if (!bBundle)
+          return;
+        for (auto aElt : aType.getElements()) {
+          auto aField = aElt.name.getValue();
+          auto bElt = bBundle.getElement(aField);
+          if (bElt) {
+            if (suffix.isTriviallyEmpty())
+              recursivePartialConnect(a, aElt.type, b, bElt.getValue().type,
+                                      aField, aFlip);
+            else
+              recursivePartialConnect(a, aElt.type, b, bElt.getValue().type,
+                                      suffix + "_" + aField, aFlip);
+          }
+        }
+      })
+      .Case<FVectorType>([&](auto aType) {
+        auto bVector = bType.dyn_cast_or_null<FVectorType>();
+        if (!bVector)
+          return;
+
+        for (size_t i = 0, e = std::min<unsigned>(aType.getNumElements(),
+                                                  bVector.getNumElements());
+             i != e; ++i) {
```

Comment:
I would put the length min in a temporary, this is pretty hard to read.  Up to you though.